### PR TITLE
Add AppFetchManifest mutation

### PR DESCRIPTION
**Pull request by korycins**
- add mutation to fetch manifest for dashboard
- tests

### CODE REVIEW

File: saleor/graphql/app/mutations.py [hunk 1]
```diff
@@ -404,3 +415,98 @@ def clean_input(cls, info, instance, data, input_cls=None):
...
+    class Meta:
+        description = "Fetch and validate manifest."
+        permissions = (AppPermission.MANAGE_APPS,)
+        error_type_class = AppError
+        error_type_field = "app_errors"
+
+    @classmethod
+    def success_response(cls, instance):
+        """Return a success response."""
+        return cls(**{"manifest": instance, "errors": []})
```
Comment (NyanKiyoshi) [inline saleor/graphql/app/mutations.py:435]: Why are we unpacking?

Comment (korycins) [reply saleor/graphql/app/mutations.py:435]: Logic taken from ModelMutation.

File: saleor/graphql/app/mutations.py [hunk 2]
```diff
@@ -404,3 +415,98 @@ def clean_input(cls, info, instance, data, input_cls=None):
...
+            msg = "Unable to fetch manifest data."
+            code = AppErrorCode.MANIFEST_URL_CANT_CONNECT.value
+            raise ValidationError({"manifest_url": ValidationError(msg, code=code)})
+        except ValueError:
+            msg = "Incorrect structure of manifest."
+            code = AppErrorCode.INVALID_MANIFEST_FORMAT.value
+            raise ValidationError({"manifest_url": ValidationError(msg, code=code)})
+        except Exception:
+            msg = "Can't fetch manifest data. Please try later."
+            code = AppErrorCode.INVALID.valuer
```
Comment (fowczarek) [inline saleor/graphql/app/mutations.py:453]:
```suggestion
            code = AppErrorCode.INVALID.value
```

File: saleor/graphql/app/mutations.py [hunk 3]
```diff
@@ -382,7 +393,7 @@ def clean_manifest_url(self, url):
             url_validator(url)
         except (ValidationError, AttributeError):
             msg = "Enter a valid URL."
-            code = AppErrorCode.INVALID.value
+            code = AppErrorCode.INVALID_MANIFEST_URL.value
```
Comment (fowczarek) [inline saleor/graphql/app/mutations.py:396]:
```
{ field: "manifest_url", code="INVALID_MANIFEST_URL" }
```
Don't provide more information than
```
{ field: "manifest_url", code="INVALID" }
```
Maybe we should use `INVALID_URL_FORMAT" or just stay with the previous version?

File: saleor/graphql/app/mutations.py (hunk 2 above)
Comment (korycins) [reply saleor/graphql/app/mutations.py:453]: 😱

**Pull request merged**